inc/zoo/map/RobinHood.h

```diff
@@ -18,9 +18,19 @@
 
 
 namespace zoo {
-
 namespace rh {
 
+struct RobinHoodException: std::runtime_error {
+    using std::runtime_error::runtime_error;
+};
+
+struct MaximumProbeSequenceLengthExceeded: RobinHoodException {
+    using RobinHoodException::RobinHoodException;
+};
+struct RelocationStackExhausted: RobinHoodException {
```

Comment:
Nit: More table size exhausted than a stack exahusted?

Reply:
Table size exhausted is actually not an exception, since the table is such that first the encoding for the PSL is exhausted.
I have to remove the sentinel part in the code.

Reply:
If we insert with a degenerate hash function that always hits the last element of the table, then we'll exhaust psl as we hit the last element of the table, so I suppose that is correct.